Realm/Realm/Linq/QueryObjects.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+
+namespace Realms
+{
+    public class QueryModel
+    {
+        public WhereClause WhereClause { get; set; }
+
+        public OrderingClause OrderingClause { get; set; }
+    }
+
+    public abstract class OrderingClause
+    {
+        public bool IsAscending { get; set; }
+
+        public bool IsReplacing { get; set; }
+
+        public string Property { get; set; }
+
+        public abstract string Kind { get; }
+    }
+
+
+    // TODO kind is not neccesarry, just the isAsc and isRep and property is needed
+    public class OrderByNode : OrderingClause
+    {
+        public override string Kind => "orderbyclause";
+    }
+
+    public class ThenByNode : OrderingClause
+    {
+        public override string Kind => "thenbyclause";
+    }
+
+    // Might be redundant
+    public class OrderByDescendingNode : OrderingClause
+    {
+        public override string Kind => "orderbydescclause";
+    }
+
+    // Might be redundant
+    public class ThenByDescendingNode : OrderingClause
+    {
+        public override string Kind => "thenbydescclause";
+    }
+
+    public class WhereClause
+    {
+        public ExpressionNode ExpNode { get; set; }
+
+        public string Kind { get; } = "whereclause";
+    }
+
+    public abstract class ExpressionNode
+    {
+        public abstract string Kind { get; }
+    }
+
+    public abstract class BooleanBinaryNode : ExpressionNode
+    {
+        public ExpressionNode Left { get; set; }
+
+        public ExpressionNode Right { get; set; }
+
+        public abstract string Operator { get; }
+    }
+
+    public class AndNode : BooleanBinaryNode
```

Comment:
Do we need to keep both the operator and the kind for these kind of nodes? It seems just one would be enough